tools/moFF/moff.xml

```diff
@@ -0,0 +1,351 @@
+                            label="Column with mass of the peptide"/>
+                        <param name="charge"
+                            type="data_column"
+                            data_ref="generic_input"
+                            label="Column with charge of ionized peptide"/>
+                    </when>
+            </conditional>
+        </xml>
+        <xml name="raw_input_macro" token_allow_multiple="true">
+            <conditional name="msms_input">
+                <param name="input_type_selector" type="select" label="Choose the format for the MS/MS file">
+                    <option value="raw">Thermo RAW file</option>
+                    <option value="mzml">mzML</option>
+                </param>
+                <when value="raw">
+                    <param argument="--inputraw" type="data" multiple="@ALLOW_MULTIPLE@" format="raw" label="RAW file(s)"
+                        help=""/>
+                </when>
+                <when value="mzml">
+                    <param argument="--inputraw" type="data" multiple="@ALLOW_MULTIPLE@" format="mzml" label="mzML file(s)"/>
+                </when>
+            </conditional>
+        </xml>
+        <!-- tokens (code snippets used in <command>) -->
+        <token name="@IDENT_INPUT_ARG_MULTIPLE@"><![CDATA[
+            ## this is where the ident input gets passed to moff/moff_all/moff_mbr
+            #if ($task.ident_input.input_type_selector == "ps"):
+                --inputtsv
+                #for $counter,$filename in enumerate($task.ident_input.psm_input):
+                    './ident_inputs/${filename.display_name}'
+            #end for
+            #else
+                --inputtsv
+                #for $counter,$filename in enumerate($task.ident_input.generic_input):
+                    './ident_inputs/${filename.display_name}'
+                #end for
+            #end if
+        ]]></token>
+        <token name="@IDENT_INPUT_ARG_SINGLE@"><![CDATA[
+            ## this is where the ident input gets passed to moff/moff_all/moff_mbr
+            #if ($task.ident_input.input_type_selector == "ps"):
+                --inputtsv './ident_inputs/${task.ident_input.psm_input.display_name}'
+            #else
+                --inputtsv './ident_inputs/${task.ident_input.generic_input.display_name}'
+            #end if
+        ]]></token>
+        <token name="@WRANGLE_IDENT_INPUT_SINGLE@"><![CDATA[
+            mkdir ./ident_inputs &&
+            #if ($task.ident_input.input_type_selector == "ps"):
+                ln -s '$task.ident_input.psm_input' './ident_inputs/$task.ident_input.psm_input.display_name' &&
+            #else
+                ## optionally remove first line
+                #if ($task.ident_input.remove_header):
+                    sed -i '1d' '$task.ident_input.generic_input' &&
+                #end if
+                ## header row with correct names: "peptide", "prot", "mod_peptide", "rt", "mz", "mass", and "charge"
```

Comment:
Is this strictly needed or can we just use metadata to describe the columns?

Reply:
Not sure on this point - I don't have much experience with metadata. But the reason I did this is because moFF requires the exact column names. Is that possible with metadata?

Reply:
Metadata is not part of the file, so if this is needed your approach is good!